components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1246,6 +1241,9 @@ where
             }
             CasualMessage::SnapshotApplied => {
                 self.fsm.has_ready = true;
+                if self.fsm.peer.destroy_after_apply_snapshot() {
+                    self.maybe_destroy();
+                }
```

Comment:
Or we can leave it unchanged and set `apply_snap_ctx` to `None` in `check_snap_status`, so that `check_stale_state` will trigger a GC message later.

Reply:
+1

Reply:
CheckApplyingSnapStatus::Idle => {
                // FIXME: It's possible that the snapshot applying task is canceled.
                // Although it only happens when shutting down the store or destroying
                // the peer, it's still dangerous if continue to handle ready for the
                // peer. So it's better to revoke `JOB_STATUS_CANCELLING` to ensure all
                // started tasks can get finished correctly.
                if self.apply_snap_ctx.is_some() {
                    return false;
                }
            }
I basically agree with the comments above----if we set apply_snap_ctx to None, the actual apply task may be still running.  It's safer to keep the whole cancelling mechanism unless we replace it completely if needed.

Reply:
Setting apply_snap_ctx to None is dangerous.  Previously, we had the assumption that once apply_snap_ctx is None, the apply task in region.rs would finish running and thus the SST file of the snapshot can be GC-ed. And if we set it None here, the apply task can still be running and Snap file GC may mistakenly delete sst files being used by the apply task.  @overvenus @Connor1996